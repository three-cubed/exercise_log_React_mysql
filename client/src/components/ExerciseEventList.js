import ExerciseEvent from './ExerciseEvent';

const ExerciseEventList = ({ exercises, restateAfterDelete }) => {
    return (
        <>
            {exercises.map((exercise) => (
                <ExerciseEvent 
                key={exercise.id} 
                id={exercise.id}
                colour={exercise.colour} 
                exerciseEventTitle={exercise.exerciseEventTitle} 
                exerciseDate={exercise.exerciseDate} 
                achievement={exercise.achievement} 
                restateAfterDelete={restateAfterDelete}
                />
            ))
            }
        </>
    )
}

export default ExerciseEventList;
