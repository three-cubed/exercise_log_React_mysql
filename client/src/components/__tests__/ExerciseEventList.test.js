import React from 'react';
import { fireEvent, render, cleanup } from '@testing-library/react';
import { toHaveStyle } from '@testing-library/jest-dom'; 
import renderer from 'react-test-renderer';
import ExerciseEventList from '../ExerciseEventList.js';

afterEach(cleanup);

let arrayOfDummyData = [
    {
        "colour":"green",
        "_id":"605ab77777",
        "exerciseEventTitle":"Mock Running",
        "exerciseDate":"12th March 2021",
        "achievement":"3 miles in 19:34"
    },
    {
        "colour":"purple",
        "_id":"605cd88888",
        "exerciseEventTitle":"Mock Weights",
        "exerciseDate":"13th March 2021"
    }
]

describe('ExerciseEventList component', () => {
    test('renders', () => {
        render(<ExerciseEventList exercises={arrayOfDummyData} restateAfterDelete={() => {}} />);
    });
    test('matches snapshot', () => {
        const tree = renderer.create(<ExerciseEventList exercises={arrayOfDummyData} restateAfterDelete={() => {}} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('shows titles from data (checked by match with original data)', () => {
        const newList = render(<ExerciseEventList exercises={arrayOfDummyData} restateAfterDelete={() => {}} />);
        expect(newList.getByText(`${arrayOfDummyData[0].exerciseEventTitle}`));
        expect(newList.getByText(`${arrayOfDummyData[1].exerciseEventTitle}`));
    });    
    test('shows titles from data (checked by required text)', () => {
        const newList = render(<ExerciseEventList exercises={arrayOfDummyData} restateAfterDelete={() => {}} />);
        expect(newList.getByText(/Running/i));
        expect(newList.getByText(/Weights/i));
    });
    test('shows dates from data', () => {
        const newList = render(<ExerciseEventList exercises={arrayOfDummyData} restateAfterDelete={() => {}} />);
        expect(newList.getByText(/12th/i));
        expect(newList.getByText(/13th/i));
    });
    test('shows acheivement from data', () => {
        const newList = render(<ExerciseEventList exercises={[arrayOfDummyData[0]]} restateAfterDelete={() => {}} />);
        expect(newList.getByText(/3 miles/i));
    });
})
