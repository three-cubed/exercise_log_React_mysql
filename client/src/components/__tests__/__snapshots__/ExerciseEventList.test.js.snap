// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseEventList component matches snapshot 1`] = `
Array [
  <div
    className="exerciseEvent"
    data-testid="exerciseEvent"
    style={
      Object {
        "borderBottom": "solid 2px",
        "borderLeft": "solid 16px",
        "color": "green",
      }
    }
  >
    <p>
        
      <small>
        12th March 2021
      </small>
      <br />
        
      <b>
        Mock Running
      </b>
          
      <br />
      <button
        className="deletionBtn"
        data-testid="deletionBtn"
        onClick={[Function]}
      >
        X
      </button>
      <span
        className="achievementSpan"
      >
          
        3 miles in 19:34
      </span>
    </p>
  </div>,
  <br />,
  <div
    className="exerciseEvent"
    data-testid="exerciseEvent"
    style={
      Object {
        "borderBottom": "solid 2px",
        "borderLeft": "solid 16px",
        "color": "purple",
      }
    }
  >
    <p>
        
      <small>
        13th March 2021
      </small>
      <br />
        
      <b>
        Mock Weights
      </b>
          
      <br />
      <button
        className="deletionBtn"
        data-testid="deletionBtn"
        onClick={[Function]}
      >
        X
      </button>
      <span
        className="achievementSpan"
      >
          
      </span>
    </p>
  </div>,
  <br />,
]
`;
